### ISSUE TITLE
Table 'ccnet_db.LDAPConfig' doesn't exist when upgrading from 6.1 to 6.3

### ISSUE THREAD

**Issue by paulocoghi** [haiwen/seafile#2062]
**Environment details:**
**SO:** Ubuntu 16.04.3 and 16.04.4
**Packages:** python2.7 libpython2.7 python-setuptools python-imaging python-ldap python-urllib3 ffmpeg python-pip python-mysqldb python-memcache python-requests
**Database:** MySQL

I was using SeaFile 6.1.2 for several months without any issues (by the way, congratulations for your incredible project!).

Today I tried to upgrade to 6.3, running:

`upgrade/upgrade_6.1_6.2.sh`
`upgrade/upgrade_6.2_6.3.sh`

The first script runs fine, but the second (from 6.2 to 6.3) returns this error:

```
Updating seafile/seahub database ...

[INFO] You are using MySQL
[INFO] updating ccnet database...
[ERROR] Failed to execute sql: (1146, "Table 'ccnet_db.LDAPConfig' doesn't exist")

Failed to upgrade your database
```

Tried in two different servers with the same results. One of these servers is a backup of the other.

**Comment by freeplant** [comment]
Upgrade to version 6.2, start seafile server. This will create the table. Then you can upgrade to version 6.3.

We will see how to solve the problem in a better way.

**Comment by paulocoghi** [comment]
Ok. As I upgraded to 6.2, I added the files of 6.2 version and started on it, and everything is running fine.

With your suggestions, I will stop everything, try to update to 6.3 and inform here the results.

**Comment by paulocoghi** [comment]
Another (probably small) issue: I stopped the 6.2 version (both `seahub` and `seafile`) and tried to upgrade to 6.3, as you said, but I received a new error:

```
seafile@sea:~/seafile/seafile-server-6.3.0$ upgrade/upgrade_6.2_6.3.sh 

-------------------------------------------------------------
This script would upgrade your seafile server from 6.2 to 6.3
Press [ENTER] to contiune
-------------------------------------------------------------



Updating seafile/seahub database ...

[INFO] You are using MySQL
[INFO] updating ccnet database...
[ERROR] Failed to execute sql: Duplicate column name 'parent_group_id'

Failed to upgrade your database

```

**Note:** the version 6.2 (6.2.5) is running perfectly fine.

**Comment by freeplant** [comment]
I'm afraid you have to upgrade the database manually by running the sql statements in upgrade/sql/6.3.0/mysql manually. Then remove contents in these files, and run the upgrade script again.

**Comment by paulocoghi** [comment]
I will try and post here the results.

**Comment by freeplant** [comment]
Fixed in 6.3.1.
